TypeError: Cannot set property 'state' of undefined

## The problem

Failures in the before hook fails the runner:

```sh
$ make google
npm install
npm WARN wdio-bugs@1.0.0 No repository field.
./node_modules/.bin/wdio --spec specs/google.js
/Users/shauvik/tmp/wdio-bugs/node_modules/webdriverio/build/lib/utils/ReporterStats.js:448
        testStats.state = 'fail';
                        ^

TypeError: Cannot set property 'state' of undefined
at ReporterStats.testFail (/Users/shauvik/tmp/wdio-bugs/node_modules/webdriverio/build/lib/utils/ReporterStats.js:448:29)
at BaseReporter.<anonymous> (/Users/shauvik/tmp/wdio-bugs/node_modules/webdriverio/build/lib/utils/BaseReporter.js:159:25)
at emitOne (events.js:96:13)
at BaseReporter.emit (events.js:188:7)
at BaseReporter.handleEvent (/Users/shauvik/tmp/wdio-bugs/node_modules/webdriverio/build/lib/utils/BaseReporter.js:300:27)
at Launcher.messageHandler (/Users/shauvik/tmp/wdio-bugs/node_modules/webdriverio/build/lib/launcher.js:609:28)
at emitTwo (events.js:106:13)
at ChildProcess.emit (events.js:191:7)
at process.nextTick (internal/child_process.js:744:12)
at _combinedTickCallback (internal/process/next_tick.js:67:7)
make: *** [google] Error 1
```

## Environment

* WebdriverIO version: latest
* Node.js version: any
* [Standalone mode or wdio testrunner](http://webdriver.io/guide/getstarted/modes.html): wdio testrunner
* if wdio testrunner, running synchronous or asynchronous tests: sync
* Additional wdio packages used (if applicable): mocha, spec reporter

## Code To Reproduce Issue [ Good To Have ]

Reproducible here: https://github.com/shauvik/wdio-bugs

This failure also leaves chromedriver and standalone selenium service running, which fails future tests.
```
MacBook-Pro-4:wdio-bugs shauvik$ ps aux | grep standalone
shauvik          60956   0.0  0.0  2434840    796 s000  S+    4:19PM   0:00.01 grep standalone
MacBook-Pro-4:wdio-bugs shauvik$ make google
npm install
npm WARN wdio-bugs@1.0.0 No repository field.
./node_modules/.bin/wdio --spec specs/google.js
/Users/shauvik/tmp/wdio-bugs/node_modules/webdriverio/build/lib/utils/ReporterStats.js:448
            testStats.state = 'fail';
                            ^

TypeError: Cannot set property 'state' of undefined
    at ReporterStats.testFail (/Users/shauvik/tmp/wdio-bugs/node_modules/webdriverio/build/lib/utils/ReporterStats.js:448:29)
    at BaseReporter.<anonymous> (/Users/shauvik/tmp/wdio-bugs/node_modules/webdriverio/build/lib/utils/BaseReporter.js:159:25)
    at emitOne (events.js:96:13)
    at BaseReporter.emit (events.js:188:7)
    at BaseReporter.handleEvent (/Users/shauvik/tmp/wdio-bugs/node_modules/webdriverio/build/lib/utils/BaseReporter.js:300:27)
    at Launcher.messageHandler (/Users/shauvik/tmp/wdio-bugs/node_modules/webdriverio/build/lib/launcher.js:609:28)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at process.nextTick (internal/child_process.js:744:12)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
make: *** [google] Error 1
MacBook-Pro-4:wdio-bugs shauvik$ ps aux | grep standalone
shauvik          61635   1.3  0.0  2434840    796 s000  S+    4:19PM   0:00.00 grep standalone
shauvik          61348   0.0  0.1  2749928   9340 s000  S     4:19PM   0:00.04 /Users/shauvik/tmp/wdio-bugs/node_modules/selenium-standalone/.selenium/chromedriver/2.25-x64-chromedriver --port=29215
shauvik          61292   0.0  0.6  8290480  94740 s000  S     4:19PM   0:01.54 /usr/bin/java -Dwebdriver.chrome.driver=/Users/shauvik/tmp/wdio-bugs/node_modules/selenium-standalone/.selenium/chromedriver/2.25-x64-chromedriver -Dwebdriver.gecko.driver=/Users/shauvik/tmp/wdio-bugs/node_modules/selenium-standalone/.selenium/geckodriver/0.11.1-x64-geckodriver -jar /Users/shauvik/tmp/wdio-bugs/node_modules/selenium-standalone/.selenium/selenium-server/2.53.1-server.jar
MacBook-Pro-4:wdio-bugs shauvik$ 
```

--- a/lib/utils/ReporterStats.js
+++ b/lib/utils/ReporterStats.js
@@ -168,12 +168,6 @@ class ReporterStats extends RunnableStats {
     getSuiteStats (runner, suiteTitle) {
         let specStats = this.getSpecStats(runner)
 
-        // The "before all" hook runs before the context of the suite is defined
-        // to prevent the creation of incorrect suite details always skip reporting for this hook
-        if (runner.title === '"before all" hook') {
-            return
-        }
-
         /**
          * if error occurs in root level hooks we haven't created any suites yet, so
          * create one here if so

